src/App.tsx

```diff
@@ -15,47 +17,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+export function getVisibleGoods(
+  goods: string[],
+  isReversed: boolean,
+  sortType: SortType,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((goodA, goodB) => {
+    switch (sortType) {
+      case SortType.ALPABET:
+        return goodA.localeCompare(goodB);
+
+      case SortType.LENGTH:
+        return goodA.length - goodB.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed
```

Comment:
I believe it is much better to separate the logic of reverse, because now we check the condition each render, even if the button was not clicked

Reply:
I have moved logic of reverse, but I still check it every render, so maybe I understood you wrong?

Reply:
you understood me correct, well done. You will get the point as soon as you get acquainted with memoization - that helps to save function without rerendering them each time if not needed.